music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,332 @@
+        self.preamble.append(f'Analyst: {self.analyst}')
+
+        if not self.proofreader:
+            self.proofreader = 'Unknown'
+        self.preamble.append(f'Proofreader: {self.proofreader}')
+
+        if self.note:  # Blank if none
+            self.preamble.append(f'Note: {self.note}')
+
+# ------------------------------------------------------------------------------
+
+    def makeMeasureStrings(self):
+        '''
+        Takes information from the Roman numerals and
+        and converts it into separate rntxt strings to print for each measure
+        stored in an dict such that analysisDict[measureNumber] = string
+        (corresponding directly to the structure for the time signatures).
+        '''
+
+        currentMeasure = -100  # i.e. fake
+        currentString = None
+        currentKey = ''
+
+        for rn in self.romanNumerals:
+            keyString = str(rn.key).split(' ')[0].replace('-', 'b')
+            if keyString != currentKey:
+                chordString = f'{keyString}: {rn.figure}'
+                currentKey = keyString
+            else:
+                chordString = str(rn.figure)
+
+            if rn.measureNumber == currentMeasure:  # continue an existing string
+                currentString = rnString(measure=rn.measureNumber,
+                                         beat=rn.beat,
+                                         chordString=chordString,
+                                         inString=currentString)
+            else:
+                self.analysisDict[currentMeasure] = currentString  # Save previous string ...
+                currentString = rnString(measure=rn.measureNumber,  # ... and start a new one.
+                                         beat=rn.beat,
+                                         chordString=chordString)
+            currentMeasure = rn.measureNumber
+
+        # Special case of last entry.
+        self.analysisDict[currentMeasure] = currentString
+
+    def prepList(self):
+        '''
+        Prepares a sequential list of text lines, integrating the timeSignatures and romanNumerals
+        from their respective dicts.
+        I.e. this prepares all lines other than the metadata preamble (handled separately)
+        '''
+
+        tsMeasures = self.timeSigMeasureDict.keys()
+
+        if not self.analysisDict:
+            self.makeMeasureStrings()
+
+        for x in range(self.firstMeasureNumber, self.lastMeasureNumber + 1):
+
+            if x in tsMeasures:  # First, before corresponding measure string
+                ts = self.timeSigMeasureDict[x]
+                self.combinedList.append(f'\nTime Signature: {ts}')
+
+            if x in self.analysisDict.keys():
+                self.combinedList.append(self.analysisDict[x])
+
+# ------------------------------------------------------------------------------
+
+    # To write
+
+    def writeRomanText(self,
+                       outPath: str = '.',
+                       fileName: str = ''):
```

Comment:
Fair enough. I'm happy to adjust if necessary, though prefer it this way given the option.

Reply:
+1 on the `fp` for the same reason @jacobtylerwalls pointed out, and because I think that some validation should be done on `fileName` if the user already provides an extension (e.g., `fileName=output.txt` should probably result in `output.txt` on the filesystem and not `output.txt.txt`). As this gets very tricky in practice, maybe `fp` is an easier solution.

Reply:
Should use `fp` like all other writer options.

One thing I've noticed with my students -- who are probably more like general music21 users -- is that more and more they're not really knowing about the encodings they're using; they just assume that an encoding is an encoding is an encoding, so the format information needs to be as much the same for each as possible.